src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -121,8 +123,23 @@ def osd_task(name, metadata, wait_for=2.0):
 @APIRouter('/osd', Scope.OSD)
 @APIDoc('OSD management API', 'OSD')
 class Osd(RESTController):
-    def list(self):
-        osds = self.get_osd_map()
+    @RESTController.MethodMap(version=APIVersion(1, 1))
+    def list(self, offset: int = 0, limit: int = 10,
+             search: str = '', sort: str = ''):
+        all_osds = self.get_osd_map()
+
+        paginator = ListPaginator(int(offset), int(limit), sort, search,
+                                  input_list=all_osds.values(),
+                                  searchable_params=['id'],
+                                  sortable_params=['id'],
+                                  default_sort='+id')
+
+        cherrypy.response.headers['X-Total-Count'] = paginator.get_count()
+        osds = {
+            key: osd
+            for key in all_osds.keys() for osd in list(paginator.list())
+            if osd['id'] == int(key)
+        }
```

Comment:
thanks @afreen23! applied your suggestion and also added a try-catch block for catching in type conversion failures. I was going to use `.isdigit()` but thought it might be better to explicitly catch the error and raise it.

Btw, the keys are always the same as the id since that is what being set in the [get_osd_map](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/controllers/osd.py#L206) method.